fsspec/implementations/reference.py

```diff
@@ -698,20 +698,20 @@ def __init__(
                 **(ref_storage_args or target_options or {}), protocol=target_protocol
             )
             ref_fs, fo2 = fsspec.core.url_to_fs(fo, **dic)
-            if ref_fs.isfile(fo2):
-                # text JSON
-                with fsspec.open(fo, "rb", **dic) as f:
-                    logger.info("Read reference from URL %s", fo)
-                    text = json.load(f)
-                self._process_references(text, template_overrides)
-            else:
+            if get_file_extension(fo2) in {"parq", "parquet"}:
```

Comment:
The path for parqut should be a directory - can we include anything finishing with "/"? Or does that get lost before making it here?

It might be better to swap the conditions around, and require .json for non-parquet? I have a feeling that should be more standard.

Reply:
> and require .json for non-parquet?

the main issue with that is that as a directory, the kerchunk "parquet" format will most likely not be compressed, while `.json` may be, and thus have a suffix. This, in turn, will make file extensions harder to extract properly, e.g. the full file extension may be  `.json.zstd`. Not sure if that's actually a problem in practice, though.